packages/field-base/src/helper-text-mixin.js

```diff
@@ -65,6 +71,33 @@ const HelperTextMixinImplementation = (superclass) =>
       this.__helperSlot.addEventListener('slotchange', this.__onHelperSlotChange.bind(this));
     }
 
+    /** @private */
+    __observeCustomHelperId(helper) {
+      this.__helperIdObserver = new MutationObserver((mutationRecord) => {
```

Comment:
Should probably add
```js
if (this.__helperIdObserver) {
  this.__helperIdObserver.disconnect()
}
```

before this line to avoid leaks.

Alternatively, we could use a single mutation observer on the host component that observes attribute changes on child nodes?

Reply:
Thanks, good catch. Moved the initialization part to `ready` to use a single observer as you suggested.

Reply:
An old `customHelper` element still doesn't get unobserved, when a new one is added. So you may also want to use
`this.__helperIdObserver.observe(this, { attributes: true, subtree: true })` here in `ready()` instead?

Reply:
lgtm now